src/components/ContentSidebar/SkillsSidebar.js

```diff
@@ -4,22 +4,163 @@
+    state: State = {
+        errors: {}
+    };
+
+    componentDidMount() {
+        const { api, file }: Props = this.props;
+        api.getMetadataAPI(false).getSkills(
+            file,
+            (cards: Array<SkillCard>) => {
+                this.setState({ cards });
+            },
+            noop
+        );
+    }
+
+    /**
+     * Updates skill metadata
+     *
+     * @private
+     * @param {string} id - File id
+     * @return {void}
+     */
+    onSave = (
+        index: number,
+        removes: Array<SkillCardEntry> = [],
+        adds: Array<SkillCardEntry> = [],
+        replaces: Array<{ replaced: SkillCardEntry, replacement: SkillCardEntry }> = []
+    ): void => {
+        const { api, file }: Props = this.props;
+        const { cards = [], errors }: State = this.state;
+        const { permissions = {} }: BoxItem = file;
+        const card = cards[index];
+        const path = `/cards/${index}`;
+        const ops: JsonPatchData = [];
+
+        if (!permissions.can_upload || !card) {
+            return;
+        }
 
-export default SkillsSidebar;
+        if (Array.isArray(replaces)) {
+            replaces.forEach(({ replaced, replacement }) => {
+                const idx = card.entries.findIndex((entry) => entry === replaced);
+                if (idx > -1) {
+                    ops.push({
+                        op: 'replace',
+                        path: `${path}/entries/${idx}`,
+                        value: replacement
+                    });
+                }
+            });
+        }
+
+        if (Array.isArray(removes)) {
+            const deletes = [];
+            removes.forEach((removed) => {
+                const idx = card.entries.findIndex((entry) => entry === removed);
+                if (idx > -1) {
+                    deletes.push(idx);
+                }
+            });
+            deletes
+                .sort()
+                .reverse()
+                .forEach((idx) => {
+                    ops.push({
+                        op: 'remove',
+                        path: `${path}/entries/${idx}`
+                    });
+                });
+        }
+
+        if (Array.isArray(adds)) {
+            adds.forEach((added) => {
+                ops.push({
+                    op: 'add',
+                    path: `${path}/entries/-`,
+                    value: added
+                });
+            });
+        }
+
+        // If no ops, don't proceed
+        if (ops.length === 0) {
+            return;
+        }
+
+        // Add test ops before any other ops
+        ops.splice(0, 0, {
+            op: 'test',
+            path,
+            value: card
+        });
+
+        api.getMetadataAPI(false).updateSkills(
+            file,
+            ops,
+            (updatedCards: Array<SkillCard>) => {
+                const clone = { ...errors };
+                delete clone[index];
+                this.setState({ cards: updatedCards, errors: clone });
```

Comment:
It looks like this PR introduces caching in the metadata API. Since you are updating something here, won't you have to update the cache as well?

Reply:
Yeah I update the cache on changes